### PR TITLE
Joe/tableview refactor

### PR DESCRIPTION
**Pull request by jmf373**
-Refactored tableviews creating enums for section types

-Created constants for reuseIdentifiers

-Cleaned up use of deprecated string splicing

### CODE REVIEW

File: Fitness/Controllers/GymDetailViewController.swift [hunk 1]
```diff
@@ -141,35 +152,41 @@ class GymDetailViewController: UIViewController, UITableViewDelegate, UITableVie
         popularTimesTitleLabel.text = "POPULAR TIMES"
         contentView.addSubview(popularTimesTitleLabel)
         
+        days = []
+        days.insert(.sunday, at: 0)
```
Comment (AAAstorga) [inline Fitness/Controllers/GymDetailViewController.swift]: you can just do `days = [.sunday, .monday, .tuesday....]`

File: Fitness/Controllers/GymDetailViewController.swift [hunk 2]
```diff
@@ -141,35 +152,41 @@ class GymDetailViewController: UIViewController, UITableViewDelegate, UITableVie
...
+        days.insert(.friday, at: 5)
+        days.insert(.saturday, at: 6)
+        
         var data: [Int]
         let date = Date()
         let todaysHours: DailyGymHours
         
-        switch date.getIntegerDayOfWeekToday() {
-        case 0:
+        switch days[date.getIntegerDayOfWeekToday() ?? 0] {
```
Comment (AAAstorga) [inline Fitness/Controllers/GymDetailViewController.swift]: is there anyway to make getIntegerDayOfWeekToday not return an optional?

File: Fitness/Controllers/GymDetailViewController.swift [hunk 3]
```diff
@@ -141,35 +152,41 @@ class GymDetailViewController: UIViewController, UITableViewDelegate, UITableVie
...
+        case .sunday:
             data = gym.popularTimesList.sunday
             todaysHours = gym.gymHours.zero ?? DailyGymHours(id: -1, dayOfWeek: 0, openTime: "", closeTime: "")
```
Comment (AAAstorga) [inline Fitness/Controllers/GymDetailViewController.swift]: why is `gym.gymHours.zero` an optional? Can we make it a non-optional to get rid of all the `??`

File: Fitness/Controllers/GymDetailViewController.swift [hunk 4]
```diff
@@ -369,76 +386,76 @@ class GymDetailViewController: UIViewController, UITableViewDelegate, UITableVie
...
             let date = Date()
             let day = (date.getIntegerDayOfWeekToday()! + indexPath.row + 1)%7
             
-            switch day{
-            case 0:
+            switch days[day]{
+            case .sunday:
                 cell.dayLabel.text = "Su"
                 if let gymHours = gym.gymHours.zero{
                     cell.hoursLabel.text = gymHours.openTime + " - " + gymHours.closeTime
```
Comment (AAAstorga) [inline Fitness/Controllers/GymDetailViewController.swift]: I believe it is cleaner if you do `cell.hoursLabel.text = "\(gymHours.openTime) - \(gymHours.closeTime)"`

File: Fitness/Controllers/GymDetailViewController.swift [hunk 5]
```diff
@@ -369,76 +386,76 @@ class GymDetailViewController: UIViewController, UITableViewDelegate, UITableVie
...
                 }else{
                     cell.hoursLabel.text = "Closed"
                 }
-            case 1:
+            case .monday:
                 cell.dayLabel.text = "M"
                 if let gymHours = gym.gymHours.one{
                     cell.hoursLabel.text = gymHours.openTime + " - " + gymHours.closeTime
```
Comment (AAAstorga) [inline Fitness/Controllers/GymDetailViewController.swift]: same as above and below

File: Fitness/Controllers/GymDetailViewController.swift (hunk 4 above)
Comment (AAAstorga) [reply Fitness/Controllers/GymDetailViewController.swift]: Also this is a lot of duplicated code. Maybe it makes sense to have a helper function like `func getHoursForGym(gym: Gym) -> String` and do the logic once in there

File: Fitness/Controllers/GymDetailViewController.swift [hunk 6]
```diff
@@ -369,76 +386,76 @@ class GymDetailViewController: UIViewController, UITableViewDelegate, UITableVie
...
         }else{
-            let cell = tableView.dequeueReusableCell(withIdentifier: "classListCell", for: indexPath) as! ClassListCell
+            let cell = tableView.dequeueReusableCell(withIdentifier: ClassListCell.identifier, for: indexPath) as! ClassListCell
             
             let gymClassInstance = todaysClasses[indexPath.row]
             
             cell.classLabel.text = gymClassInstance.classDescription.name
             cell.timeLabel.text = gymClassInstance.startTime
             
             if (cell.timeLabel.text?.hasPrefix("0"))!{
```
Comment (AAAstorga) [inline Fitness/Controllers/GymDetailViewController.swift]: no force unwrapping

File: Fitness/Controllers/GymDetailViewController.swift [hunk 7]
```diff
@@ -369,76 +386,76 @@ class GymDetailViewController: UIViewController, UITableViewDelegate, UITableVie
...
-                cell.timeLabel.text = cell.timeLabel.text?.substring(from: String.Index(encodedOffset: 1))
+                cell.timeLabel.text = String((cell.timeLabel.text?[String.Index(encodedOffset: 1)...])!)
```
Comment (AAAstorga) [inline Fitness/Controllers/GymDetailViewController.swift]: no force unwrapping

File: Fitness/Controllers/GymDetailViewController.swift [hunk 8]
```diff
@@ -453,56 +470,58 @@ class GymDetailViewController: UIViewController, UITableViewDelegate, UITableVie
...
             let date = Date()
-            let currentDay = date.getIntegerDayOfWeekToday()
+            let currentDay = days[date.getIntegerDayOfWeekToday() ?? 0]
             
+            // TODO : check this code after refectoring/linking with backend, gymHours format may changes
             switch currentDay{
-            case 0:    
+            case .sunday:
                 if let gymHours = gym.gymHours.zero{
                     header.hoursLabel.text = gymHours.openTime + " - " + gymHours.closeTime
```
Comment (AAAstorga) [inline Fitness/Controllers/GymDetailViewController.swift]: same as above, abstract all of this into a helper function

File: Fitness/Controllers/GymDetailViewController.swift [hunk 9]
```diff
@@ -453,56 +470,58 @@ class GymDetailViewController: UIViewController, UITableViewDelegate, UITableVie
...
+            case .saturday:
                 if let gymHours = gym.gymHours.six{
                     header.hoursLabel.text = gymHours.openTime + " - " + gymHours.closeTime
                 }else{
                     header.hoursLabel.text = "Closed Today"
                 }
-            default:
-                header.hoursLabel.text = ""
+            }
+            if (header.hoursLabel.text?.hasPrefix("0"))!{
```
Comment (AAAstorga) [inline Fitness/Controllers/GymDetailViewController.swift]: no force unwrap

File: Fitness/Controllers/GymDetailViewController.swift [hunk 10]
```diff
@@ -453,56 +470,58 @@ class GymDetailViewController: UIViewController, UITableViewDelegate, UITableVie
...
+                header.hoursLabel.text = String((header.hoursLabel.text?[String.Index(encodedOffset: 1)...])!)
```
Comment (AAAstorga) [inline Fitness/Controllers/GymDetailViewController.swift]: no force unwrap

File: Fitness/Controllers/HomeController.swift [hunk 11]
```diff
@@ -8,12 +8,19 @@
...
+    case lookingFor
+}
+
 class HomeController: UIViewController, UITableViewDataSource, UITableViewDelegate {
     
     // MARK: - INITIALIZATION
     var tableView: UITableView!
     var statusBarBackgroundColor: UIView!
     
+    var sections = [SectionType]()
```
Comment (AAAstorga) [inline Fitness/Controllers/HomeController.swift]: you should be doing `var sections: [SectionType] = []`. same for the others after this as well

File: Fitness/Controllers/GymDetailViewController.swift (hunk 3 above)
Comment (jmf373) [reply Fitness/Controllers/GymDetailViewController.swift]: It was initially an optional to work around the fact that backend didn't provide hours for days that gyms weren't open (I realize now that backend should have handled this on their end), but with the new use of GraphQL this shouldn't be an issue, and if it is I'll talk with backend and have them make the changes on their end.

File: Fitness/Controllers/ClassListViewController.swift [hunk 12]
```diff
@@ -161,14 +161,14 @@ class ClassListViewController: UITableViewController, UISearchBarDelegate {
...
     
     override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCell(withIdentifier: "classListCell", for: indexPath) as! ClassListCell
+        let cell = tableView.dequeueReusableCell(withIdentifier: ClassListCell.identifier, for: indexPath) as! ClassListCell
         
         if let gymClassInstance = validGymClassInstances?[indexPath.row]{
             
             cell.classLabel.text = gymClassInstance.classDescription.name
             cell.timeLabel.text = gymClassInstance.startTime
             if (cell.timeLabel.text?.hasPrefix("0"))!{
```
[the remark this answers is not in the document]
Comment (AAAstorga) [reply Fitness/Controllers/ClassListViewController.swift]: I was delusional, ignore

**Pull request merged**